pkg/runtime/pubsub/subscriptions.go

```diff
@@ -270,7 +270,13 @@ func declarativeFile(resourcesPath string, log logger.Logger) (subs []Subscripti
 
 		bytesArray := bytes.Split(b, []byte("\n---"))
 		for _, item := range bytesArray {
-			subs, err = appendSubscription(subs, item)
+			// Skip empty items
+			item = bytes.TrimSpace(item)
+			if len(item) == 0 {
+				continue
```

Comment:
@ItalyPaleAle Can you add tests to cover this scenario?

Reply:
Added